services/pull/pull.go

```diff
@@ -546,6 +546,42 @@ func (errs errlist) Error() string {
 	return ""
 }
 
+// RetargetChildrenOnMerge retarget children pull requests on merge if possible
+func RetargetChildrenOnMerge(ctx context.Context, doer *user_model.User, pr *issues_model.PullRequest) error {
+	if setting.Repository.PullRequest.RetargetChildrenOnMerge && pr.BaseRepoID == pr.HeadRepoID {
+		return RetargetBranchPulls(ctx, doer, pr.HeadRepoID, pr.HeadBranch, pr.BaseBranch)
+	}
+	return nil
+}
+
+// RetargetBranchPulls change target branch for all pull requests who's base branch is the branch
+func RetargetBranchPulls(ctx context.Context, doer *user_model.User, repoID int64, branch, targetBranch string) error {
```

Comment:
Doesn't this method belong semantically into `models`?

Reply:
Why so? IMO this is a helper method that should live nearby e.g. CloseBranchPulls

Reply:
@delvh I tried, this isn't possible because of an import cycle issue when importing "modules" inside "modules/pull". I also don't see a reason to do so, honestly....

Reply:
@delvh, the logic is a little bit more complex when just updating db. `ChangeTargetBranch` is used for retargeting and it is (and was) located exactly in ths same file.